share/github-backup-utils/ghe-backup-repositories-cluster-ng

```diff
@@ -0,0 +1,379 @@
+
++ /*/??/
++ /*/??/??/
++ /*/??/??/??/
++ /*/??/??/??/gist/
++ /*/??/??/??/gist/*.git
+- /*/??/??/??/gist/*.git/objects
+- /*/??/??/??/gist/*.git/refs
+- /*/??/??/??/gist/*.git/packed-refs
+- /*/??/??/??/gist/*.git/logs
++ /*/??/??/??/gist/*.git/**
+
++ /*/nw/??/??/??/
++ /*/nw/??/??/??/*/
++ /*/nw/??/??/??/*/*.git
+- /*/nw/??/??/??/*/*.git/objects
+- /*/nw/??/??/??/*/*.git/refs
+- /*/nw/??/??/??/*/*.git/packed-refs
+- /*/nw/??/??/??/*/*.git/logs
++ /*/nw/??/??/??/*/*.git/**
+RULES
+
+    # Sync packed refs files. This is performed before sync'ing loose refs since
+    # loose refs trump packed-refs information.
+  echo 1>&3
+  echo "* Transferring packed-refs files ..." 1>&3
+  rsync_repository_data $1:122 $2 -z <<RULES
+- /__*__/
+- /info/
+
++ /*/
++ /*/*.git
++ /*/*.git/packed-refs
+
++ /*/??/
++ /*/??/??/
++ /*/??/??/??/
++ /*/??/??/??/gist/
++ /*/??/??/??/gist/*.git
++ /*/??/??/??/gist/*.git/packed-refs
+
++ /*/nw/??/
++ /*/nw/??/??/
++ /*/nw/??/??/??/
++ /*/nw/??/??/??/*/
++ /*/nw/??/??/??/*/*.git
++ /*/nw/??/??/??/*/*.git/packed-refs
+RULES
+    # Sync loose refs and reflogs. This must be performed before object data is
+    # transferred to ensure that all referenced objects are included.
+  echo 1>&3
+  echo "* Transferring refs and reflogs ..."  1>&3
+  rsync_repository_data $1:122 $2 -z <<RULES
+- /__*__/
+- /info/
+
++ /*/
++ /*/*.git
++ /*/*.git/refs
++ /*/*.git/refs/**
++ /*/*.git/logs
++ /*/*.git/logs/**
+
++ /*/??/
++ /*/??/??/
++ /*/??/??/??/
++ /*/??/??/??/gist/
++ /*/??/??/??/gist/*.git
++ /*/??/??/??/gist/*.git/refs
++ /*/??/??/??/gist/*.git/refs/**
++ /*/??/??/??/gist/*.git/logs
++ /*/??/??/??/gist/*.git/logs/**
+
++ /*/nw/??/
++ /*/nw/??/??/
++ /*/nw/??/??/??/
++ /*/nw/??/??/??/*/
++ /*/nw/??/??/??/*/*.git
++ /*/nw/??/??/??/*/*.git/refs
++ /*/nw/??/??/??/*/*.git/refs/**
++ /*/nw/??/??/??/*/*.git/logs
++ /*/nw/??/??/??/*/*.git/logs/**
+RULES
+
+    # Sync git objects and pack files. Compression is disabled during this phase
+    # since these files are already well compressed.
+  echo 1>&3
+  echo "* Transferring objects and packs ..." 1>&3
+  rsync_repository_data $1:122 $2 -H <<RULES
+- /__*__/
+- /info/
+
++ /*/
++ /*/*.git
++ /*/*.git/objects
+- /*/*.git/objects/**/tmp_*
++ /*/*.git/objects/**
+
++ /*/??/
++ /*/??/??/
++ /*/??/??/??/
++ /*/??/??/??/gist/
++ /*/??/??/??/gist/*.git
++ /*/??/??/??/gist/*.git/objects
+- /*/??/??/??/gist/*.git/objects/**/tmp_*
++ /*/??/??/??/gist/*.git/objects/**
+
++ /*/nw/??/
++ /*/nw/??/??/
++ /*/nw/??/??/??/
++ /*/nw/??/??/??/*/
++ /*/nw/??/??/??/*/*.git
++ /*/nw/??/??/??/*/*.git/objects
+- /*/nw/??/??/??/*/*.git/objects/**/tmp_*
++ /*/nw/??/??/??/*/*.git/objects/**
+RULES
+
+ echo 1>&3
+
+}
+
+# rsync all the repositories
+bm_start "$(basename $0) - Repo sync"
+for file_list in $tempdir/*.rsync; do
+  hostname=$(basename $file_list .rsync)
+
+  repo_num=$(cat $file_list | wc -l)
+  echo "* Transferring $repo_num repositories from $hostname"
+
+  sync_data $hostname $file_list &
```

Comment:
this N seems like it should be small? Like only 3 or 4 processes given the file lists right?